samcli/commands/sync/command.py

```diff
@@ -403,10 +410,39 @@ def execute_code_sync(
         else set(get_all_resource_ids(stacks))
     )
 
+    sync_function_event_mapper = {
```

Comment:
Is there some way that we can avoid using maps? Reason is we will need to maintain these maps in the future whenever there's a new sync flow

Reply:
I'm not sure whether there's a way to avoid it completely, but there may be a way to refactor the maps to be simpler. Initially I used multiple `if .. elif` statements, but pylint didn't seem too fond of it lol